### PR TITLE
Feature/batch endpoint

### PR DESCRIPTION
**Pull request by mephenor**
This PR includes functionality to query dataset information and the version is bumped to 1.1.0 from 1.0.1.

## Changes
- A new GET endpoint `dataset_information/{dataset_id}` is exposed to obtain information on all files within a dataset
- Datasets are created and updated by listening to events following the `MetadataDatasetOverview` schema
- Datasets are deleted by listening to events following the `MetadataDatasetID` schema
- Dataset and file information creation and deletion is decoupled as these events are created from different sources
- Dataset information returns some information on all files in a dataset in all cases. If the corresponding file information is registered, full file information is populated in the part of the response for the corresponding file. If the file information is not registered yet or deleted, only the accession is returned to acknowledge that the file is/was part of the dataset.
- Consistent use of the term accession instead of ID within the code core. Boundary uses a mix of ID and accession, depending on the incoming event schema

## Open questions
~~- Should the 1.0.1 state be ported separately as base for this PR?
Would require a moderate amount of menial work to adapt to the base template and rebase onto that state later on~~
~~Done in a [separate PR](https://github.com/ghga-de/dataset-information-service/pull/2), needs to be merged first~~ Done

### CODE REVIEW

File: src/dins/core/information_service.py [hunk 1]
```diff
@@ -31,53 +36,121 @@ class InformationService(InformationServicePort):
...
+            dataset = await self._dataset_dao.get_by_id(dataset_id)
+            log.debug("Found mapping for dataset %s.", dataset_id)
+        except ResourceNotFoundError as error:
+            dataset_not_found = self.DatasetNotFoundError(dataset_accession=dataset_id)
+            log.warning(dataset_not_found)
+            raise dataset_not_found from error
+
+        file_information: list[FileAccession | FileInformation] = []
+
+        for file_accession in sorted(dataset.file_accessions):
```
Comment (TheByronHimes) [inline src/dins/core/information_service.py:137]: The file accessions field is already a list and should therefore have a definite sequence. Did you run into problems with it returning different sequences here?

**Review by TheByronHimes — CHANGES_REQUESTED** [review]

**Review by TheByronHimes — CHANGES_REQUESTED** [review]
Most comments are formatting, naming, or typo-related, no real conceptual problems.

File: src/dins/adapters/inbound/fastapi_/http_exceptions.py [hunk 2]
```diff
@@ -19,6 +19,28 @@
...
+
+        dataset_id: str
+
+    def __init__(self, *, dataset_id: str, status_code: int = 404):
+        """Construct message and init the exception."""
+        super().__init__(
+            status_code=status_code,
+            description=(
+                f"Information for the dataset with ID {
+                    dataset_id} is not registered."
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/dins/adapters/inbound/fastapi_/http_exceptions.py:38]: yep

File: src/dins/core/information_service.py (hunk 1 above)
Comment (mephenor) [reply src/dins/core/information_service.py:137]: Not sure if the accessions in the incoming event that populates the database are always sorted. 
Wanted to make sure they always will be for this endpoint.

File: src/dins/adapters/inbound/fastapi_/routes.py [hunk 3]
```diff
@@ -25,16 +25,29 @@
...
+        ),
+        "model": http_exceptions.HttpDatasetNotFoundError.get_body_model(),
+    },
     "fileInformation": {
         "description": (
-            "A configuration or external communication error has occurred and details "
-            + "should not be communicated to the client"
+            "File information consisting of file id, sha256 checksum of the unencrypted"
+            "file content and file size of the unencrypted file in bytes.",
         ),
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/dins/adapters/inbound/fastapi_/routes.py:45]: In general, not really.
But the service has incoming data where both are used.
As this is not directly in the backend file service domain and cosumed by the frontend, I wanted to stick with accession internally (where possible) and use both in the comments to denote that they are indeed interchangeable.
Apparently missed that one

File: src/dins/adapters/inbound/fastapi_/routes.py [hunk 4]
```diff
@@ -44,22 +57,52 @@
...
+        raise http_exceptions.HttpDatasetNotFoundError(dataset_id=dataset_id) from error
+
+    return http_responses.HttpDatasetInformationResponse(
+        dataset_information=dataset_file_information
+    )
+
+
 @router.get(
     "/file_information/{file_id}",
     summary="Return public file information for the given file id, i.e. public accession.",
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/dins/adapters/inbound/fastapi_/routes.py:101]: It's id/accession now

File: src/dins/core/information_service.py [hunk 5]
```diff
@@ -31,53 +36,121 @@ class InformationService(InformationServicePort):
...
+            # Only log if information to be inserted is a mismatch
+            if existing_mapping != dataset_file_accessions:
+                await self._dataset_dao.update(dataset_file_accessions)
+                log.info("Updated existing dataset %s.", dataset_accession)
+        except ResourceNotFoundError:
+            await self._dataset_dao.insert(dataset_file_accessions)
+            log.debug(
+                "Successfully inserted file accession mapping for dataset %s.",
+                dataset_accession,
+            )
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/dins/core/information_service.py:94]: There was a reason once, but I think I refactored some stuff since, so easier solution it is.

**Review by TheByronHimes — APPROVED** [review]

**Pull request merged**